plugins/parsers/xpath/parser.go

```diff
@@ -126,37 +129,20 @@ func (p *Parser) Parse(buf []byte) ([]telegraf.Metric, error) {
 }
 
 func (p *Parser) ParseLine(line string) (telegraf.Metric, error) {
-	t := time.Now()
 
-	switch len(p.Configs) {
+	metrics, err := p.Parse([]byte(line))
+	if err != nil {
+		return nil, err
+	}
+
+	switch len(metrics) {
 	case 0:
 		return nil, nil
 	case 1:
-		config := p.Configs[0]
-
-		doc, err := p.document.Parse([]byte(line))
-		if err != nil {
-			return nil, err
-		}
-
-		selected := doc
-		if len(config.Selection) > 0 {
-			selectedNodes, err := p.document.QueryAll(doc, config.Selection)
-			if err != nil {
-				return nil, err
-			}
-			if len(selectedNodes) < 1 || selectedNodes[0] == nil {
-				p.debugEmptyQuery("metric selection", doc, config.Selection)
-				return nil, fmt.Errorf("cannot parse line with empty selection")
-			} else if len(selectedNodes) != 1 {
-				return nil, fmt.Errorf("cannot parse line with multiple selected nodes (%d)", len(selectedNodes))
-			}
-			selected = selectedNodes[0]
-		}
-
-		return p.parseQuery(t, doc, selected, config)
+		return metrics[0], nil
+	default:
+		return metrics[0], fmt.Errorf("cannot parse line with multiple (%d) metrics", len(metrics))
 	}
-	return nil, fmt.Errorf("cannot parse line with multiple (%d) configurations", len(p.Configs))
```

Comment:
I want to be certain I understand the change to `ParseLine`: this is removing all the length selection code, returning the first value, and all the empty selection logic is moved out of here?

Reply:
Yup, discussed with @srebhan in Slack

Reply:
@powersj the idea before was to have only one `config` section for a line parser (therefore we can have restrictive length tests). However, people might see a non-homogeneous stream of XMLs not all having the same structure. Therefore, the user should be able to specify multiple `config` sections (e.g. one per XML format). With allowing empty selections (for all non-matching XML formats) this is now possible and the only restriction for the "line-parser" should be to return no or a single metric (to fulfill the interface).